File: Components/Strata.Client.Core/main/typescript/Service/StandardResponse.ts
import {IResponse} from "./IResponse";

export
class StandardResponse
    implements IResponse
{
    private readonly requestPath: string;
    private readonly response: Response;

    constructor(rootPath: string,methodPath: string,response: Response)
    {
        this.requestPath = rootPath + '/' + methodPath;
        this.response = response;
    }

    getRequestPath(): string
    {
        return this.requestPath;
    }

    getMediaType(): string
    {
        if (this.response.headers.has("Content-Type"))
            return this.response.headers.get("Content-Type");

        return "unknown";
    }

    getStatus(): number
    {
        return this.response.status;
    }

    getStatusReason(): string
    {
        return this.response.statusText;
    }

    async readEntity<E>(): Promise<E>
    {
        return await
            this
                .response
                .json()
                .then((entity) => entity as E);
    }

    close(): void
    {
    }
}

File: Components/Strata.Client.Core/src/main/java/strata/client/core/service/StandardResponse.java
//////////////////////////////////////////////////////////////////////////////
// StandardResponse.java
//////////////////////////////////////////////////////////////////////////////

package strata.client.core.service;


import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.MultivaluedMap;
import jakarta.ws.rs.core.Response;

import java.util.Map;

public
class StandardResponse
    implements IResponse
{
    private final String   itsRequestPath;
    private final Response itsResponse;

    public
    StandardResponse(String rootPath,String methodPath,Response response)
    {
        itsRequestPath = rootPath + "/" + methodPath;
        itsResponse = response;
    }

    @Override
    public String
    getRequestPath()
    {
        return itsRequestPath;
    }

    @Override
    public MediaType
    getMediaType()
    {
        return itsResponse.getMediaType();
    }

    @Override
    public Response.Status
    getStatusAsEnum()
    {
        return itsResponse.getStatusInfo().toEnum();
    }

    @Override
    public int
    getStatusAsInt()
    {
        return itsResponse.getStatus();
    }

    @Override
    public String
    getStatusReason()
    {
        return itsResponse.getStatusInfo().getReasonPhrase();
    }

    @Override
    public MultivaluedMap<String,Object>
    getHeaders()
    {
        return itsResponse.getHeaders();
    }

    @Override
    public <E> E
    readEntity(Class<E> entityType)
    {
        return itsResponse.readEntity(entityType);
    }

    @Override
    public void
    close()
    {
        itsResponse.close();
    }
}

//////////////////////////////////////////////////////////////////////////////
